### PR TITLE
Add troubleshooting section into readme, add self-healing option for eol endings

### PR DESCRIPTION
**Pull request by vtcaregorodtcev**
Sometimes, building docker container on windows stops working, because of wrong file endings. And because ui is only part of that repository, it's not good enough to configure git hooks or something similar. 

So I suppose, providing self healing option to fix wrong file, is enough to get started quickly.

### CODE REVIEW

File: ui/src/App.tsx [hunk 1]
```diff
@@ -22,7 +22,7 @@ const RedirectToFirstPaper = () => {
     useEffect(() => {
         getAllocatedPaperStatus().then((allocation) => {
             const first = allocation.papers[0];
-            setSha(first.sha);
+            if (first) setSha(first.sha);
```
Comment (codeviking) [inline ui/src/App.tsx:25]: This is a functional change which seems unrelated to the other changes in this PR. Is it intentional?

File: README.md [hunk 2]
```diff
@@ -182,6 +182,16 @@ If you find PAWLS helpful for your research, please consider cite PAWLS.
...
 
+### Troubleshooting
+
+1. Windows EOL format (CRLF) vs Linux (LF)
+
+Sometimes you may wondering why building docker image is not working and `ui` part is taking down. It's because some files were saved in incorrect format. To healing such files just go to the repo's root and paste following in terminal.
+
+```bash
+~ (cd ./ui && yarn && yarn lint:fix) # with brackets, to stay in same directory
+```
```
Comment (codeviking) [inline README.md:193]: Interesting.  The program is usually executed in a Docker container, so I wouldn't expect this to be an issue.

Are you trying to run the program on a Windows machine outside of Docker? Or are you still hitting this while using `docker-compose`?

File: ui/src/App.tsx (hunk 1 above)
Comment (vtcaregorodtcev) [reply ui/src/App.tsx:25]: So, this is throwing an error when we trying to start app first time and didn't put some pdf's into skiff folder. Should I create separate PR with this fix and implementation of case described above (show some page that says "there is no pdf's")

File: README.md (hunk 2 above)
Comment (vtcaregorodtcev) [reply README.md:193]: So, probably case was reproduced on older version of pawls. 

when we tried to `docker compose build` on windows 10, it thrown following issues.
MicrosoftTeams-image (4)

File: ui/src/App.tsx (hunk 1 above)
Comment (codeviking) [reply ui/src/App.tsx:25]: > Should I create separate PR with this fix and implementation of case described above (show some page that says "there is no pdf's" or something like this)

Yea, that'd be great (and seems like a good addition). Thank you for taking this on!

File: README.md (hunk 2 above)
Comment (codeviking) [reply README.md:193]: Interesting, I'm going to try to investigate a bit, just to make sure we don't inadvertently introduce a compatibility issue that impacts linux clients.

Thanks for your patience.

File: ui/src/App.tsx (hunk 1 above)
Comment (codeviking) [reply ui/src/App.tsx:25]: I think you can revert this change, now that your other PR has merged. Thx!

File: ui/.eslintrc.js [hunk 3]
```diff
@@ -1,3 +1,6 @@
 module.exports = {
     extends: ['@allenai/eslint-config-varnish'],
+    rules: {
+        'eol-last': ['error', 'always'],
+    },
```
Comment (codeviking) [inline ui/.eslintrc.js:5]: This is a good addition. Thanks!

File: README.md [hunk 4]
```diff
@@ -182,6 +182,16 @@ If you find PAWLS helpful for your research, please consider cite PAWLS.
 }
 ```
 
+### Troubleshooting
+
+1. Windows EOL format (CRLF) vs Linux (LF)
```
Comment (codeviking) [inline README.md:187]:
```suggestion
#### Windows EOL format (CRLF) vs Linux (LF)
```

File: README.md (hunk 2 above)
Comment (codeviking) [reply README.md:193]: Did the application actually fail to start in this scenario? My hope is that the application still starts, even though the linter is failing.

It's interesting that the linter in this case prefers the line-endings of the host OS, not of the container it's running in. That said I don't know a lot about running docker on Windows, as I don't use Windows :).

File: README.md [hunk 5]
```diff
@@ -182,6 +182,16 @@ If you find PAWLS helpful for your research, please consider cite PAWLS.
...
+
+Sometimes you may wondering why building docker image is not working and `ui` part is taking down. It's because some files were saved in incorrect format. To healing such files just go to the repo's root and paste following in terminal.
```
Comment (codeviking) [inline README.md:189]:
```suggestion
The application was developed for Linux, and might fail to start on Windows because of line-ending differences.

To fix this, run this command from the root of the repository:
```

File: README.md [hunk 6]
```diff
@@ -182,6 +182,16 @@ If you find PAWLS helpful for your research, please consider cite PAWLS.
...
+
+```bash
+~ (cd ./ui && yarn && yarn lint:fix) # with brackets, to stay in same directory
```
Comment (codeviking) [inline README.md:192]:
```suggestion
~ (cd ./ui && yarn && yarn lint:fix) # with parenthesis, to stay in same directory
```

**Review by codeviking — COMMENTED** [review]
I left a few suggested edits to the verbiage. Let me know what you think.

**Review by codeviking — APPROVED** [review]

File: README.md (hunk 2 above)
Comment (vtcaregorodtcev) [reply README.md:195]: How I might guess, all linter rules are configured to throw errors when code is not fit them and due to that fact building process interrupted.

Comment (codeviking) [reply README.md:195]: Ah, yup. And we execute a build before starting the application locally, even though we don't need to.

Thanks, that's super helpful. I'm going to make a quick tweak that should fix that and in doing so make the local development environment a bit faster to start. Thanks!

Comment (codeviking) [reply README.md:195]: This PR will at least allow the container to start when there's formatting errors;
https://github.com/allenai/pawls/pull/134

Thanks for catching this!
